src/librustc/query/mod.rs

```diff
@@ -70,19 +70,19 @@ rustc_queries! {
             eval_always
         }
 
-        // An HIR item with a `DefId` that can own other HIR items which do not themselves have
+        // An HIR node with a `DefId` that can own other HIR nodes which do not themselves have
```

Comment:
These comments turned out to be inaccurate since some nodes have both a `DefId` and a `HirId` without being a HIR owner themselves. Maybe fix them?

Reply:
> These comments turned out to be inaccurate since some nodes have both a `DefId` and a `HirId` without being a HIR owner themselves. Maybe fix them?

Right, currently only a subset of `DefId`s are HIR owners. I'll take a second look at these comments to try and make that clear.

Reply:
Tried something, but whew that phrasing is hard to use.

I would be happier with `hir::Owner` (or I guess `hir::OwnerKind`) being a subset of `Node` and explicitly listing all of the supported node types.

Then it sort of self-documents what kinds of nodes are owners.
And I could also use it to add a `visit_owner` method to HIR visitors that would work with exhaustive matching.

But I didn't want to put any changes like that into this tiny PR.

Reply:
I think it's probably better to assume that the reader know what a HIR owner is here. I can barely make sense of the comments and I wrote the code =P